File: c#/IntegerPalindrome.cs
using System;

public class Solution
{
	public bool IntergerPalindrome(int integer)
	{
		int exponent = 0;
		if(integer < 0) return false;
		if(integer > 0) exponent = (int)Math.Log10(integer);

		while(integer != 0)
		{
			if(integer % 10 != (int)(integer / (int)Math.Pow(10, exponent))) return false;
			integer = integer % (int)Math.Pow(10, exponent);
			integer = (int)(integer / 10);
			exponent -= 2;
		}

		return true;
	}
}

public class LeetCode
{
	public static void Main()
	{
		Solution solution = new Solution();
		Console.WriteLine(solution.IntergerPalindrome(121));
	}
}

File: java/IntegerPalindrome.java
class Solution
{
	public boolean IsIntergerPalindrome(int x)
	{
		int n = 0;
		if(x < 0) return false;
		if(x > 0) n = (int)Math.log10(x);

		while(x != 0)
		{
			if(x % 10 != (int)(x / (int)Math.pow(10, n))) return false;
			x = x % (int)Math.pow(10, n);
			x = (int)(x / 10);
			n -= 2;
		}

		return true;
	}
}

public class Easy_Is_Integer_Palindrome
{
	public static void main(String[] args)
	{
		Solution solution = new Solution();
		System.out.println(solution.IsIntergerPalindrome(121));
	}
}
